src/pycram/language.py

```diff
@@ -260,34 +261,50 @@ def __init__(self, condition: Union[Callable, Fluent] = None):
         """
         super().__init__(None, None)
         self.kill_event = threading.Event()
+        self.exception_queue = queue.Queue()
         if callable(condition):
             self.condition = Fluent(condition)
         elif isinstance(condition, Fluent):
             self.condition = condition
         else:
             raise AttributeError("The condition of a Monitor has to be a Callable or a Fluent")
 
-    def perform(self):
+    def perform(self) -> Tuple[State, Any]:
         """
         Behavior of the Monitor, starts a new Thread which checks the condition and then performs the attached language
         expression
 
-        :return: The result of the attached language expression
+        :return: The state of the attached language expression, as well as a list of the results of the children
         """
         def check_condition():
-            while not self.condition.get_value() and not self.kill_event.is_set():
-                time.sleep(0.1)
-            if self.kill_event.is_set():
-                return
-            for child in self.children:
-                child.interrupt()
+            while not self.kill_event.is_set():
+                try:
+                    cond = self.condition.get_value()
+                    if cond:
+                        for child in self.children:
+                            if hasattr(child, 'interrupt'):
+                                child.interrupt()
```

Comment:
yes it is intended, but its not a decision that i have made. RetryMonitor uses the already existing [Monitor](https://pycram.readthedocs.io/en/latest/notebooks/language.html#monitor) that was added by (i think) Jonas. here is the Doc for the monitor: 
"Monitor allows to monitor the execution of a language expression and interrupt it as soon as a given condition is fulfilled. The condition can either be a Callable which returns a boolean or a Fluent. When executed the Monitor will create a separate thread which will check if the condition is satisfied with a frequency of 10 Hz. If the condition is satisfied the execution of the language expression will be interrupted."

>  Can I understand a condition to be equivalent to a failure state, that wants to be avoided?
kind of? i mean it depends on the context. for example in suturo, they used monitoring to detect when a plate was in the correct position when placing it in the dishwasher. for this they monitored the force put onto the wrist of the hsr, and when it exceeded a certain value, it would mean that the plate is in the correct position, and the movement needs to be interrupted now. but i would not call that a failurestate.

Reply:
done